fileutil/perm_other.go

```diff
@@ -0,0 +1,43 @@
+// +build !windows
+
+package fileutil
+
+import (
+	"os"
+	"os/user"
+	"strconv"
+	"syscall"
+)
+
+func HasPermissionToDir(info os.FileInfo) (bool, error) {
+	st, _ := info.Sys().(*syscall.Stat_t)
+	if st == nil {
+		return true, nil
+	}
+
+	perm := info.Mode().Perm()
+	user, err := user.Current() // TODO: do this at start-up
```

Comment:
Another point in favor of keeping this inside `interp` - we can more easily hook into the internal `Runner` state, such as an initialised `os/user.User`.

Reply:
Seems that `os/user` already caches the user struct, so no need to worry about that.

Reply:
I was thinking more along the lines of getting the value at startup, not on the first `cd`. But I guess it can't change anyway, so feel free to just remove the TODO.